middle_end/semantics_of_primitives.ml

```diff
@@ -23,8 +23,8 @@ let for_primitive (prim : Clambda_primitives.primitive) =
   match prim with
   | Pmakeblock _
   | Pmakearray (_, Mutable) -> Only_generative_effects, No_coeffects
-  | Pmakearray (_, Immutable) -> No_effects, No_coeffects
-  | Pduparray (_, Immutable) ->
+  | Pmakearray (_, (Immutable | Immutable_unique)) -> No_effects, No_coeffects
+  | Pduparray (_, (Immutable | Immutable_unique)) ->
```

Comment:
Not related to this PR per se, but why are immutable array creation not considered to have generative effects ?

Reply:
They are considered to have generative effects in flambda2. Here apparently `Only_generative_effects` corresponds to flambda2's `Only_genertive_effects Mutable`.